rs_zktrie/src/db.rs

```diff
@@ -1,13 +1,15 @@
 use crate::raw::ImplError;
 use std::collections::HashMap;
+use std::rc::Rc;
+
 pub trait ZktrieDatabase: Clone {
     fn put(&mut self, k: Vec<u8>, v: Vec<u8>) -> Result<(), ImplError>;
-    fn get(&self, k: &[u8]) -> Result<Vec<u8>, ImplError>;
+    fn get(&self, k: &[u8]) -> Result<Rc<[u8]>, ImplError>;
 }
 
 #[derive(Clone, Default)]
 pub struct SimpleDb {
-    db: HashMap<Vec<u8>, Vec<u8>>,
+    db: HashMap<Box<[u8]>, Rc<[u8]>>,
```

Comment:
Is there any reference to indicate that a `Box<[u8]>` has much better effecient than `Vec<u8>` in hash map?

Reply:
`Box<[u8]>` is better than `Vec<u8>` for immutable usage. `Box<[u8]>` is only a single fat pointer.